File: javascript/5637.js
const input = require("fs").readFileSync("/dev/stdin").toString();

const words = [...input.matchAll(/[a-zA-Z-]+/g)].map((match) => match[0]);
const longest = Math.max(...words.map((w) => w.length));

console.log(words.find((w) => w.length === longest).toLowerCase());

File: c#/5637.cs
using System;
using System.Text;
using System.Text.RegularExpressions;

class Program
{
    static void Main()
    {
        StringBuilder builder = new();
        string line;

        while ((line = Console.ReadLine()) != null)
        {
            builder.Append(line + "\n");
        }

        Regex regex = new(@"[a-zA-Z-]+", RegexOptions.Compiled);
        var words = regex.Matches(builder.ToString()).Select(m => m.Value);
        int longest = words.Select(w => w.Length).Max();

        Console.WriteLine(words.First(w => w.Length == longest).ToLower());
    }
}
